src/compile/scale/domain.ts

```diff
@@ -166,85 +160,89 @@ function parseSingleChannelDomain(scaleType: ScaleType, domain: Domain, model: U
     } else {
       if (isDateTime(domain[0])) {
         return (domain as DateTime[]).map((dt) => {
-          return {signal: dateTimeExpr(dt, true)};
+          return {signal: `{data: ${dateTimeExpr(dt, true)}}`};
         });
       }
-      return domain;
+      return [domain];
     }
   }
 
   const stack = model.stack;
   if (stack && channel === stack.fieldChannel) {
     if(stack.offset === 'normalize') {
-      return [0, 1];
+      return [[0, 1]];
     }
-    return {
-      data: model.requestDataName(MAIN),
-      fields: [
-        model.field(channel, {suffix: 'start'}),
-        model.field(channel, {suffix: 'end'})
-      ]
-    };
+
+    const data = model.requestDataName(MAIN);
+    return [{
```

Comment:
It now becomes an array of domains, rather than a single Vega domain object. 

In fact, I don't think you really need to do this (but rather modify the merge method to handle this case.)  That said, if you're lazy, at least rename the method.